actions/Network.js

```diff
@@ -28,16 +28,19 @@ function createWretchError(status, message) {
   return error;
 }
 
-function wretchWithOptions(url, shouldRetry = false) {
+function wretchWithOptions(url, options = {
+  shouldRetry: false,
+  requestTimeout: REQUEST_TIMEOUT, // in seconds
+}) {
   return wretch(url)
-    .middlewares(shouldRetry ? [retry()] : [])
+    .middlewares(options.shouldRetry ? [retry()] : [])
     .addon(AbortAddon())
-    .resolve((resolver) => resolver.setTimeout(REQUEST_TIMEOUT))
+    .resolve((resolver) => resolver.setTimeout(options.requestTimeout * 1000))
     .resolve((resolver) => {
       return resolver.fetchError((error) => {
         if (error.name === 'AbortError') {
           logger.error('Request aborted', error);
-          throw createWretchError(408, 'Request timed out');
+          throw createWretchError(504, 'Gateway Timeout');
```

Comment:
I spoke with the AppBuilder team, and we agreed that a 504 status code is more appropriate for this use case. See https://cq-dev.slack.com/archives/C013UDBFBGB/p1715356714149289